src/composables/secureStorageRef.ts

```diff
@@ -0,0 +1,146 @@
+import { Ref, ref, watch } from 'vue';
+import { validateMnemonic } from '@aeternity/bip39';
+
+import type { StorageKey } from '@/types';
+import { decrypt, encrypt, watchUntilTruthy } from '@/utils';
+import { IS_MOBILE_APP, STORAGE_KEYS } from '@/constants';
+import { WalletStorage } from '@/lib/WalletStorage';
+import { useAccounts } from './accounts';
+import { type ICreateStorageRefOptions, useStorageRef } from './storageRef';
+
+/**
+ * Create a secure storage ref that encrypts and decrypts the value using the password.
+ * Is a wrapper around the useStorageRef composable.
+ */
+export function useSecureStorageRef<T = string | object | any[]>(
+  initialState: T,
+  storageKey: StorageKey,
+  options: Omit<
+    ICreateStorageRefOptions<T>,
+    'serializer' | 'isSecure' | 'onBackgroundSync'
+  > = {},
+) {
+  const isWriting = ref(false);
+
+  async function getDecryptedValue(encryptedValue: any) {
+    if (IS_MOBILE_APP || !encryptedValue) {
+      return encryptedValue;
+    }
+    watchUntilTruthy(() => !isWriting.value);
+    const { passwordKey } = useAccounts();
+    await watchUntilTruthy(passwordKey);
```

Comment:
1. I suggest to move the initial `useAccounts` line to the beginning of `useSecureStorageRef`.
2. Maybe the accounts composable could have a separate method `getPasswordKey` which would use the `watchUntilTruthy`. I'm suggesting this because similar code is already used in line 53.